core/src/main/java/hudson/model/ItemGroupMixIn.java

```diff
@@ -142,10 +142,14 @@ public synchronized TopLevelItem createTopLevelItem( StaplerRequest req, Stapler
         TopLevelItem result;
 
         String requestContentType = req.getContentType();
-        if(requestContentType==null)
+        String mode = req.getParameter("mode");
+        if (requestContentType == null
+                && !(mode != null && mode.equals("copy")))
             throw new Failure("No Content-Type header set");
 
-        boolean isXmlSubmission = requestContentType.startsWith("application/xml") || requestContentType.startsWith("text/xml");
+        boolean isXmlSubmission = requestContentType != null
```

Comment:
As @KostyaSha mentioned in Skype, this formatting change is not related. IMO it can be accepted, because this part of the code is not under the active development now